class Solution {
    public int maxSubArray(int[] nums) {
        int n = nums.length;
        int maxSum = Integer.MIN_VALUE;
        int curSum = 0;

        for(int i = 0; i < n; i++) {
            if(curSum < 0) {
                curSum = 0;
            }

            curSum = curSum + nums[i];

            maxSum = Math.max(maxSum, curSum);
        }

        return maxSum;
    }
}